src/client/runner/automation/playback/move.js

```diff
@@ -1,355 +1,364 @@
+
+        this.endPoint = null;
+
+        // moving state
+        this.movingTime = null;
+        this.x          = null;
+        this.y          = null;
+        this.startTime  = null;
+        this.endTime    = null;
+        this.distanceX  = null;
+        this.distanceY  = null;
+    }
+
+    static onMoveToIframeRequest (e) {
+        var iframePoint = {
+            x: e.message.endX,
+            y: e.message.endY
+        };
+
+        var iframeWin                   = e.source;
+        var iframe                      = domUtils.getIframeByWindow(iframeWin);
+        var iframeBorders               = styleUtils.getBordersWidth(iframe);
+        var iframePadding               = styleUtils.getElementPadding(iframe);
+        var iframeRectangle             = positionUtils.getIFrameClientCoordinates(iframe);
+        var iframePointRelativeToParent = positionUtils.getIframePointRelativeToParentFrame(iframePoint, iframeWin);
+
+        var cursorPosition    = Cursor.getPosition();
+        var intersectionPoint = positionUtils.isInRectangle(cursorPosition, iframeRectangle) ? cursorPosition :
+                                getLineRectIntersection(cursorPosition, iframePointRelativeToParent, iframeRectangle);
+
+        var intersectionRelatedToIframe = {
+            x: intersectionPoint.x - iframeRectangle.left,
+            y: intersectionPoint.y - iframeRectangle.top
+        };
+
+        var moveOptions = new MoveOptions();
+
+        moveOptions.modifiers = e.message.modifiers;
+        moveOptions.offsetX   = intersectionRelatedToIframe.x + iframeBorders.left + iframePadding.left;
+        moveOptions.offsetY   = intersectionRelatedToIframe.y + iframeBorders.top + iframePadding.top;
+
+        var moveAutomation = new MoveAutomation(iframe, moveOptions);
+
+        moveAutomation
+            .run()
+            .then(() => {
+                Cursor.setActiveWindow(iframeWin);
+
+                var responseMsg = {
+                    cmd: MOVE_RESPONSE_CMD,
+                    x:   intersectionRelatedToIframe.x,
+                    y:   intersectionRelatedToIframe.y
                 };
 
-                //NOTE: mousemove event can't be simulated on the point when cursor was at the start. Therefore we increases
-                // a minimal distance 1 px.
-                if (currentCursorPosition.x === startX && currentCursorPosition.y === startY) {
-                    if (inSelect) {
-                        movingStepCallback();
-                        return;
-                    }
+                messageSandbox.sendServiceMsg(responseMsg, iframeWin);
+            });
+    }
 
-                    if (distanceX !== 0)
-                        currentCursorPosition.x = currentCursorPosition.x + (distanceX > 0 ? 1 : -1);
-                    else if (distanceY !== 0)
-                        currentCursorPosition.y = currentCursorPosition.y + (distanceY > 0 ? 1 : -1);
+    static onMoveFromIframeRequest (e) {
```

Comment:
Naming is not quite clear. What's the difference between `onMoveToIframeRequest` and `onMoveFromIframeRequest `. \cc @VasilyStrelyaev for the assistance